src/components/interface/cbufp/cbuf_c.c

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2010 by The George Washington University.  All rights reserved.
+ *
+ * Redistribution of this file is permitted under the GNU General
+ * Public License v2.
+ *
+ * Author: Gabriel Parmer, gparmer@gwu.edu, 2010
+ * Updated by Qi Wang and Jiguo Song, 2011
+ * Updated and simplified by removing sub-page allocations, Gabe Parmer, 2012
+ */
+
+#include <cos_component.h>
+#include <print.h>
+#include <cbuf.h>
+#include <cbufp.h>
+#include <cbuf_vect.h>
+#include <cos_vect.h>
+#include <cos_debug.h>
+#include <cos_alloc.h>
+#include <valloc.h>
+
+#define CSLAB_ALLOC(sz)   alloc_page()
+#define CSLAB_FREE(x, sz) free_page(x)
+#include <cslab.h>
```

Comment:
You aren't using slabs any more (for freelist node allocation), so you should really remove this.